crates/supervisor/core/src/reorg/task.rs

```diff
@@ -0,0 +1,397 @@
+use crate::SupervisorError;
+use alloy_eips::BlockNumHash;
+use alloy_primitives::{B256, ChainId};
+use alloy_rpc_client::RpcClient;
+use alloy_rpc_types_eth::Block;
+use derive_more::Constructor;
+use kona_supervisor_storage::{DbReader, StorageRewinder};
+use std::sync::Arc;
+use tracing::{debug, info, trace, warn};
+
+/// Handles reorg for a single chain
+#[derive(Debug, Constructor)]
+pub(crate) struct ReorgTask<DB> {
+    chain_id: ChainId,
+    db: Arc<DB>,
+    rpc_client: RpcClient,
+}
+
+impl<DB> ReorgTask<DB>
+where
+    DB: DbReader + StorageRewinder + Send + Sync + 'static,
+{
+    /// Processes reorg for a single chain
+    pub(crate) async fn process_chain_reorg(&self) -> Result<(), SupervisorError> {
+        // Find last valid source block for this chain
+        let rewind_target_source = match self.find_rewind_target().await? {
+            Some(source) => source,
+            None => {
+                // No need to re-org for this chain
+                return Ok(());
+            }
+        };
```

Comment:
```suggestion
        let Some(rewind_target_source) = match self.find_rewind_target().await? else {
            // No need to re-org for this chain
            return Ok(());
        };
```
when the `None` variant results in a return statement, this syntax is preferred